test/e2e/branding_test.go

```diff
@@ -0,0 +1,164 @@
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/testframework"
+)
+
+// Test the customization-branding features of the config map
+func setupBrandingTestCase(t *testing.T) *testframework.Clientset {
+	client := testframework.MustNewClientset(t, nil)
+	return client
+}
+
+// the simpler test first would be branding set via the operator config
+func TestOperatorConfigBranding(t *testing.T) {
+	validBrands := []operatorsv1.Brand{operatorsv1.BrandOKD, operatorsv1.BrandOCP, operatorsv1.BrandOnline, operatorsv1.BrandDedicated}
+	client := setupBrandingTestCase(t)
+	// Save the original config and restore after test
+	originalConfig, err := client.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("could not get operator config, %v", err)
+	}
+	defer client.Consoles().Update(originalConfig)
+
+	for _, expectedBrand := range validBrands {
+		// helper to update the operator config
+		setOperaterConfigBrand(t, client, expectedBrand)
+
+		// now check if it has set the brand
+		gotBrand := getOperaterConfigBrand(t, client)
+		if expectedBrand != gotBrand {
+			t.Fatalf("brand expected %v but brand got was %v", expectedBrand, gotBrand)
+		}
+	}
+
+}
+
+func TestBrandingFromManagedConfigMap(t *testing.T) {
+	client := setupBrandingTestCase(t)
+	validBrands := []operatorsv1.Brand{operatorsv1.BrandOKD, operatorsv1.BrandOCP, operatorsv1.BrandOnline, operatorsv1.BrandDedicated}
+
+	t.Logf("update data for the config map in openshift-config-managed namespace")
```

Comment:
Lets pull the log output into the range loop to see which brands are we updating. Check the example beneath...